src/algebra/geom_sum.lean

```diff
@@ -28,7 +28,8 @@ $\sum_{i=0}^{n-1} x^i y^{n-1-i}$ and variants thereof.
 * `geom_sum_Ico` proves that $\sum_{i=m}^{n-1} x^i=\frac{x^n-x^m}{x-1}$ in a division ring.
 * `geom_sum₂_Ico` proves that $\sum_{i=m}^{n-1} x^i=\frac{x^n-y^{n-m}x^m}{x-y}$ in a field.
 
-Several variants are recorded, generalising the results to semirings.
+Several variants are recorded, generalising the results to semirings. In this case, the assumption
+of commutativity of the ring is replaced by `commute x y`.
```

Comment:
I think you're confusing `comm_ring`, `ring`, `semiring`, and `comm_semiring` here. The generalization from `comm_*` to `*` is done using `commutes`.

I have no idea how the generalization from `*` to `semi_*` is done here, but this sentence suggests it is also by `commutes` which is wrong.

Reply:
I'm not confusing it, but I agree that it might be confusing to some readers. I'll change it.

Reply:
Hopefully it is clearer now.